Sources/Swift/Helper/SentrySwiftSerialization.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+@objcMembers
+class SentrySwiftSerialization: NSObject {
+    
+    private static let SENTRY_BAGGAGE_MAX_SIZE = 8_192
+    
+    static func baggageEncodedDictionary(_ dictionary: [String: String]) -> String {
+        var items: [String] = []
+        items.reserveCapacity(dictionary.count)
+        
+        var allowedSet = CharacterSet.alphanumerics
+        allowedSet.insert(charactersIn: "-_.")
+        var currentSize = 0
+        
+        for (key, value) in dictionary {
+            let keyDescription = key.addingPercentEncoding(withAllowedCharacters: allowedSet) ?? ""
+            let valueDescription = value.addingPercentEncoding(withAllowedCharacters: allowedSet) ?? ""
+            let item = "\(keyDescription)=\(valueDescription)"
+            if item.count + currentSize <= SENTRY_BAGGAGE_MAX_SIZE {
+                currentSize += item.count + 1 // +1 is to account for the comma that will be added for each extra item
+                items.append(item)
+            }
```

Comment:
h: An empty key or value should not be allowed, there is not added value.

Reply:
Good catch. A test should fail for this 😉

Reply:
Yes, I believe we should add it. 
Maybe the previous understanding was that those methods don't return nil, and that is the cause of the error.

Reply:
That makes sense. I update the code.